### ISSUE TITLE
Can create Federation without CDN

### ISSUE THREAD

**Issue by dandypham** [apache/trafficcontrol#4052]
## I'm submitting a ...

- [x] bug report
- [ ] new feature / enhancement request
- [ ] improvement request (usability, performance, tech debt, etc.)
- [ ] other 

## Traffic Control components affected ...

- [ ] CDN in a Box
- [ ] Documentation
- [ ] Grove
- [ ] Traffic Control Client
- [ ] Traffic Monitor
- [x] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] unknown

## Current behavior:

I can create a Federation without create a CDN first.
First do a POST to the API with some random number after (/cdns/):
`https://{{TO_BASE_URL}}/api/{{api_version}}/cdns/999/federations`
With this json payload:
```json
{
        "cname": "cdntest.",
        "ttl": 60,
        "userIds": [2],
        "description": "A test federation"
}
```
It will be success with status 200.

## Expected / new behavior:

The API shouldn’t allow to create Federations without create a CDN first.

**Comment by ocket8888** [comment]
That URI ought to result in a 404 if the CDN ID doesn't exist.

But also, a 200 status code doesn't technically mean that it was a real success. Can you confirm if in this case the Federation was actually created (e.g. via a `GET` request to `/cdns/999/federations` or with direct database access)?

**Comment by dandypham** [comment]
Yes, I just checked again. So you can create a Federations without the CDN. Then the Federation is not display as first but if you give it a Delivery Service. It will fall into whatever the CDN the Delivery Service connect to.
So the steps are:
Create a Federation with non existence cdnsID: 
POST `https://{{TO_BASE_URL}}/api/{{api_version}}/cdns/999/federations`

With this json payload:
```json
{
        "cname": "cdntest.",
        "ttl": 60,
        "userIds": [2],
        "description": "A test federation"
}
```
Add a Delivery Service to it:
POST `https://{{TO_BASE_URL}}/api/{{api_version}}/federations/{{federationsID}}/deliveryservice`
With this json payload:
```json
 {
        "dsIds": [14],
        "replace": true
}
```
Then the Federations will display:
GET `https://{{TO_BASE_URL}}/api/{{api_version}}/federations/{{federationsID}}/deliveryservice`

**Comment by ocket8888** [comment]
That's interesting. That means that it'll change the CDN of the Federation whenever it's assigned to a DS. Undocumented side-effect.

**Comment by dandypham** [comment]
Yes, so if the user decide to change the Delivery Service it will change the CDN also.

**Comment by mitchell852** [comment]
Yes, this is a very odd endpoint as it completely ignores the cdn id route parameter in `POST api/{{api_version}}/cdns/999/federations`

Putting the federation "in a cdn" requires associating the federation with a delivery service that is in that cdn.

So I guess a federation can be in multiple cdns as a federation can have multiple delivery services assigned to it via `POST api/{{api_version}}/federations/{{federationsID}}/deliveryservices`

But in practice, I don't think it makes any sense to have a federation assigned to multiple delivery services. 

There are many problems with these federation endpoints that i would classify as "poor design" and not as "bugs".

**Comment by mitchell852** [comment]
I'm removing the bug label as it's not a bug per se but rather poor api design. Creating a federation via

`POST api/{{api_version}}/cdns/999/federations`

completely ignores the cdn Id route parameter
